File: converted/classes_only/TraitFactory$Trait.java
package zombie.characters.traits;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Locale;
import zombie.characters.skills.PerkFactory.Perk;
import zombie.characters.skills.PerkFactory.Perks;
import zombie.core.textures.Texture;
import zombie.debug.DebugLog;
import zombie.interfaces.IListBoxItem;

public class TraitFactory$Trait implements IListBoxItem {
   public String traitID;
   public String name;
   public int cost;
   public String description;
   public boolean prof;
   public Texture texture = null;
   private boolean removeInMP = false;
   private List<String> freeRecipes = new ArrayList<>();
   public ArrayList<String> MutuallyExclusive = new ArrayList<>(0);
   public HashMap<Perk, Integer> XPBoostMap = new HashMap<>();

   public void addXPBoost(Perk var1, int var2) {
      if (var1 != null && var1 != Perks.None && var1 != Perks.MAX) {
         this.XPBoostMap.put(var1, var2);
      } else {
         DebugLog.General.warn("invalid perk passed to Trait.addXPBoost trait=%s perk=%s", new Object[]{this.name, var1});
      }
   }

   public List<String> getFreeRecipes() {
      return this.freeRecipes;
   }

   public void setFreeRecipes(List<String> var1) {
      this.freeRecipes = var1;
   }

   public TraitFactory$Trait(String var1, String var2, int var3, String var4, boolean var5, boolean var6) {
      this.traitID = var1;
      this.name = var2;
      this.cost = var3;
      this.description = var4;
      this.prof = var5;
      this.texture = Texture.getSharedTexture("media/ui/Traits/trait_" + this.traitID.toLowerCase(Locale.ENGLISH) + ".png");
      if (this.texture == null) {
         this.texture = Texture.getSharedTexture("media/ui/Traits/trait_generic.png");
      }

      this.removeInMP = var6;
   }

   public String getType() {
      return this.traitID;
   }

   public Texture getTexture() {
      return this.texture;
   }

   public String getLabel() {
      return this.name;
   }

   public String getLeftLabel() {
      return this.name;
   }

   public String getRightLabel() {
      int var1 = this.cost;
      if (var1 == 0) {
         return "";
      } else {
         String var2 = "+";
         if (var1 > 0) {
            var2 = "-";
         } else if (var1 == 0) {
            var2 = "";
         }

         if (var1 < 0) {
            var1 = -var1;
         }

         return var2 + new Integer(var1).toString();
      }
   }

   public int getCost() {
      return this.cost;
   }

   public boolean isFree() {
      return this.prof;
   }

   public String getDescription() {
      return this.description;
   }

   public void setDescription(String var1) {
      this.description = var1;
   }

   public ArrayList<String> getMutuallyExclusiveTraits() {
      return this.MutuallyExclusive;
   }

   public HashMap<Perk, Integer> getXPBoostMap() {
      return this.XPBoostMap;
   }

   public boolean isRemoveInMP() {
      return this.removeInMP;
   }

   public void setRemoveInMP(boolean var1) {
      this.removeInMP = var1;
   }
}

File: DoNotUse/include/zombie/characters/traits/TraitFactory$Trait.cpp
#pragma once
#include <string>
#include <vector>
#include <memory>
#include <unordered_map>
#include <unordered_set>
#include <cstdint>
#include "zombie/characters/skills/PerkFactory/Perk.h"
#include "zombie/characters/skills/PerkFactory/Perks.h"
#include "zombie/core/textures/Texture.h"
#include "zombie/debug/DebugLog.h"
#include "zombie/interfaces/IListBoxItem.h"
#include <algorithm>

namespace zombie {
namespace characters {
namespace traits {


class TraitFactory {
public:
    std::string traitID;
    std::string name;
    int cost;
    std::string description;
    bool prof;
    Texture texture = nullptr;
    bool removeInMP = false;
   private List<std::string> freeRecipes = std::make_unique<std::vector<>>();
   public std::vector<std::string> MutuallyExclusive = new std::vector<>(0);
   public std::unordered_map<Perk, int> XPBoostMap = std::make_unique<std::unordered_map<>>();

    void addXPBoost(Perk var1, int var2) {
      if (var1 != nullptr && var1 != Perks.None && var1 != Perks.MAX) {
         this.XPBoostMap.put(var1, var2);
      } else {
         DebugLog.General.warn("invalid perk passed to Trait.addXPBoost trait=%s perk=%s", new Object[]{this.name, var1});
      }
   }

   public List<std::string> getFreeRecipes() {
      return this.freeRecipes;
   }

    void setFreeRecipes(List<std::string> var1) {
      this.freeRecipes = var1;
   }

   public TraitFactory$Trait(std::string var1, std::string var2, int var3, std::string var4, boolean var5, boolean var6) {
      this.traitID = var1;
      this.name = var2;
      this.cost = var3;
      this.description = var4;
      this.prof = var5;
      this.texture = Texture.getSharedTexture("media/ui/Traits/trait_" + this.traitID.toLowerCase(Locale.ENGLISH) + ".png");
      if (this.texture == nullptr) {
         this.texture = Texture.getSharedTexture("media/ui/Traits/trait_generic.png");
      }

      this.removeInMP = var6;
   }

    std::string getType() {
      return this.traitID;
   }

    Texture getTexture() {
      return this.texture;
   }

    std::string getLabel() {
      return this.name;
   }

    std::string getLeftLabel() {
      return this.name;
   }

    std::string getRightLabel() {
    int var1 = this.cost;
      if (var1 == 0) {
         return "";
      } else {
    std::string var2 = "+";
         if (var1 > 0) {
            var2 = "-";
         } else if (var1 == 0) {
            var2 = "";
         }

         if (var1 < 0) {
            var1 = -var1;
         }

         return var2 + std::make_shared<int>(var1);
      }
   }

    int getCost() {
      return this.cost;
   }

    bool isFree() {
      return this.prof;
   }

    std::string getDescription() {
      return this.description;
   }

    void setDescription(const std::string& var1) {
      this.description = var1;
   }

   public std::vector<std::string> getMutuallyExclusiveTraits() {
      return this.MutuallyExclusive;
   }

   public std::unordered_map<Perk, int> getXPBoostMap() {
      return this.XPBoostMap;
   }

    bool isRemoveInMP() {
      return this.removeInMP;
   }

    void setRemoveInMP(bool var1) {
      this.removeInMP = var1;
   }
}
} // namespace traits
} // namespace characters
} // namespace zombie
